CallOptionsTest.withDeadlineNanoTime is flaky

```
io.grpc.CallOptionsTest > withDeadlineNanoTime FAILED
    java.lang.AssertionError: <-25128839 ns from now> and <-1322031 ns from now> should have been within <20000000ns> of each other
        at com.google.common.truth.FailureStrategy.fail(FailureStrategy.java:24)
        at com.google.common.truth.FailureStrategy.fail(FailureStrategy.java:20)
        at com.google.common.truth.Subject.failWithRawMessage(Subject.java:381)
        at io.grpc.testing.DeadlineSubject.access$400(DeadlineSubject.java:52)
        at io.grpc.testing.DeadlineSubject$1.of(DeadlineSubject.java:81)
        at io.grpc.CallOptionsTest.withDeadlineNanoTime(CallOptionsTest.java:161)
```

As seen (may have been restarted):
https://travis-ci.org/grpc/grpc-java/jobs/121289248
https://travis-ci.org/grpc/grpc-java/jobs/121487416
(<-33620603 ns from now> and <-2319629 ns from now> should have been within <20000000ns> of each other)

--- a/core/src/test/java/io/grpc/CallOptionsTest.java
+++ b/core/src/test/java/io/grpc/CallOptionsTest.java
@@ -152,13 +152,17 @@ public class CallOptionsTest {
   @Test
   @SuppressWarnings("deprecation")
   public void withDeadlineNanoTime() {
-    CallOptions opts = CallOptions.DEFAULT.withDeadlineNanoTime(System.nanoTime());
+    // Create Deadline near calling System.nanoTime to reduce clock differences
+    Deadline reference = Deadline.after(-1, NANOSECONDS);
+    long rawDeadline = System.nanoTime() - 1;
+    CallOptions opts = CallOptions.DEFAULT.withDeadlineNanoTime(rawDeadline);
     assertThat(opts.getDeadlineNanoTime()).isNotNull();
+    // This is not technically correct, since nanoTime is permitted to overflow, but the chances of
+    // that impacting this test are very remote.
     assertThat(opts.getDeadlineNanoTime()).isAtMost(System.nanoTime());
     assertThat(opts.getDeadline().isExpired()).isTrue();
 
-    assertAbout(deadline()).that(opts.getDeadline())
-        .isWithin(20, MILLISECONDS).of(Deadline.after(0, SECONDS));
+    assertAbout(deadline()).that(opts.getDeadline()).isWithin(50, MILLISECONDS).of(reference);
   }
 
   // TODO(carl-mastrangelo): consider making a CallOptionsSubject for Truth.

